File: iosApp/iosApp/Note.swift
//
//  Created by Amsavarthan Lv on 03/12/22.
//  Copyright © 2022 orgName. All rights reserved.
//

import shared

extension Note: Identifiable {

    static let emptyNote = Note(
            id: nil,
            title: "Untitled",
            content: "",
            created: DateTimeUtil().now()
    )

    func copy(id: KotlinLong? = nil, title: String? = nil, content: String? = nil) -> Note {
        Note(id: id ?? self.id, title: title ?? self.title, content: content ?? self.content, created: self.created)
    }

}

File: shared/src/commonMain/kotlin/com/amsavarthan/note/domain/note/Note.kt
package com.amsavarthan.note.domain.note

import com.amsavarthan.note.domain.CommonParcelable
import com.amsavarthan.note.domain.CommonParcelize
import com.amsavarthan.note.domain.CommonTypeParceler
import com.amsavarthan.note.domain.LocalDateTimeParceler
import com.amsavarthan.note.domain.time.DateTimeUtil
import kotlinx.datetime.LocalDateTime
import kotlinx.datetime.serializers.LocalDateTimeIso8601Serializer
import kotlinx.serialization.Serializable

@Serializable
@CommonParcelize
data class Note(
    val id: Long? = null,
    val title: String = "Untitled",
    val content: String = "",
    @Serializable(with = LocalDateTimeIso8601Serializer::class)
    @CommonTypeParceler<LocalDateTime, LocalDateTimeParceler>()
    val created: LocalDateTime = DateTimeUtil.now()
) : CommonParcelable